### PR TITLE
Add delete and update API

### PR DESCRIPTION
**Pull request by YayunFan**
est Cases for Recommendations
- Create a recommendation and add it to the database
- Create a recommendation and assert that it exists
- Delete a Recommendation
- Test deserialization of a Recommendation
- Test deserialization of bad data
- Test deserialization of a Recommendation with missing data
- Test deserialization of no data
- Find Recommendation by customer_id
- Find Recommendation by product_id
- Find Recommendation by recommend_type
- Find a Recommendation by ID
- Test serialization of a Recommendation
- Update a Recommendation

Recommendation Server Tests
- Create a new Recommendation
- Create a new Recommendation with bad content type
- Delete a Recommendation
- Get a single Recommendation
- Get a list of Recommendations
- Get a Recommendation thats not found
- Update an existing Recommendation
- Update a Recommendation thats not found

Name                  Stmts   Miss  Cover
-----------------------------------------
service/__init__.py      18      0   100%
service/models.py        57      0   100%
service/service.py       68      0   100%
-----------------------------------------
TOTAL                   143      0   100%
----------------------------------------------------------------------
Ran 21 tests in 1.564s

OK

### CODE REVIEW

File: service/service.py [hunk 1]
```diff
@@ -80,6 +80,43 @@ def create_recommendations():
...
+def update_recommendations(rec_id):
+    """
+    Update a Recommendations
+    This endpoint will update a Recommendation based the body that is posted
+    """
+    app.logger.info('Request to update recommendation with id: %s', rec_id)
+    check_content_type('application/json')
+    recommendation = Recommendation.find(rec_id)
+    if not recommendation:
+        raise NotFound("Recommendation with id '{}' was not found.".format(rec_id))
```
Comment (cheryl1223) [inline service/service.py:115]: I think this line is not covered in the test, could you please write a test for that? By the way you can run `coverage report -m` to see which lines are missed.
I know you may think this is trivial but it's a good practice to cover all branches.

Comment (hi348) [reply service/service.py]: Is it okay to cover two issues with one pull request? I think for now we can just re-purpose the existing "update" issue and it should be fine right?

**Comment by hi348** [comment]
Is it okay to cover two issues with one pull request? I think for now we can just re-purpose the existing "update" issue and it should be fine right? Let me know if this is acceptable and I'll probably move that into the icebox or new issues pipeline.

**Comment by cheryl1223** [comment]
> Is it okay to cover two issues with one pull request? I think for now we can just re-purpose the existing "update" issue and it should be fine right? Let me know if this is acceptable and I'll probably move that into the icebox or new issues pipeline.

I think it's a good practice to have one PR for one issue.

File: service/service.py (hunk 1 above)
Comment (YayunFan) [reply service/service.py:115]: Thank you for your comment! Have added another test.

**Comment by YayunFan** [comment]
I will cover only one issue in the next pull request. Thank you for your comments!

**Pull request merged**